packages/docs/src/stories/templates/header.stories.ts

```diff
@@ -97,11 +97,25 @@ export const SampleA = {
         </sd-navigation-item>
       </nav>
     </sd-drawer>
-    <script>
-      function openDrawerSampleA() {
+    <script type="module">
+      await Promise.all([
+        customElements.whenDefined('sd-navigation-item'),
+        customElements.whenDefined('sd-drawer')
+      ]).then(() => {
         const drawerSampleA = document.getElementById('sample-a-drawer');
-        drawerSampleA.show();
-      }
+        const navigationItemSampleA = document.getElementById('open-menu-sample-a');
+        const buttonSampleA = navigationItemSampleA.shadowRoot.querySelector('button');
+
+        buttonSampleA.setAttribute('aria-controls', 'sample-a-drawer');
+        //Add the necessary ARIA attributes to prevent only being added after action
+        buttonSampleA.setAttribute('aria-expanded', 'false');
+        drawerSampleA.addEventListener('sd-hide', () => buttonSampleA.setAttribute('aria-expanded', 'false'));
+
+        navigationItemSampleA.addEventListener('click', () => {
+          drawerSampleA.show();
+          buttonSampleA.setAttribute('aria-expanded', 'true');
```

Comment:
This irritates me. Why do we have to set something in the ShadowDOM from outside? Feels like double work?

Reply:
I think you missed the point here: The navigation-item should get the  additional aria attributes here in this template.

Reply:
Hey @mariohamann, the navigation-item is ignored and it is not part of the accessibility tree, so what actually is there is the HTML element button, that is inside the shadowDOM. This means that adding aria-attributes to the navigation-item is useless because it won't be readed or announced by screen readers.

![image](https://github.com/user-attachments/assets/4a56dfeb-e0d5-409e-a37d-7fa859c658fb)

Reply:
The same happens for the close button in the drawer. The sd-button used there is ignored and to have the screen reader announcement the aria atributes need to be added to the HTML element 'button' inside, that is why I set something in the ShadowDOM from outside. It looks ugly and it is not perfect but I was't able to find a better solution that doesn't compromisses accessibility. If you have a proposal for a better approach please let me know.